src/API/Reports/Customers/DataStore.php

```diff
@@ -75,7 +75,7 @@ protected function assign_report_columns() {
 			'date_last_order'  => "MAX( {$wpdb->prefix}wc_order_stats.date_created ) as date_last_order",
 			'orders_count'     => 'SUM( CASE WHEN parent_id = 0 THEN 1 ELSE 0 END ) as orders_count',
 			'total_spend'      => 'SUM( total_sales ) as total_spend',
-			'avg_order_value'  => '( SUM( total_sales ) / COUNT( order_id ) ) as avg_order_value',
+			'avg_order_value'  => 'CASE WHEN SUM( CASE WHEN parent_id = 0 THEN 1 ELSE 0 END ) = 0 THEN NULL ELSE SUM( total_sales ) / SUM( CASE WHEN parent_id = 0 THEN 1 ELSE 0 END ) END AS avg_order_value',
```

Comment:
What do you think about putting the `CASE` statement for `orders_count` into a variable that can be reused here?

Reply:
> What do you think about putting the CASE statement for orders_count into a variable that can be reused here?

I would prefer to do that. I wasn't sure if `orders_count` could be used in other summary calculations in all the supported versions of MySQL. eg.

```
CASE WHEN orders_count <= 0 THEN NULL ELSE SUM( total_sales ) / orders_count END AS avg_order_value
```

Reply:
I mean a PHP variable, like in the [gross sales calculation](https://github.com/woocommerce/woocommerce-admin/blob/master/src/API/Reports/Orders/Stats/DataStore.php#L80-L86).